### PR TITLE
Fix relative links remapping in docs

### LINKED ISSUE #182: Rebuild a website and deploy a new version
## 🐛 Bug Report

404 Errors on the website should have been fixed by https://github.com/fastify/fastify/pull/2434 yet those errors still present on the website. Thus, it would be nice to rebuild the website and deploy a new version, which will fix those things.

## To Reproduce

* Go to the documentation on Server.decorate: https://www.fastify.io/docs/latest/Server/#decorate
* Then click on the here hyperlink, it should redirect you to https://www.fastify.io/docs/latest/Server/Decorators.md
* Receive 404 Content not found
and many others

## Expected behavior

A user should be redirected to the correct page where they can read the information and see code examples.

## Your Environment

- *os*: Mac

### PR DESCRIPTION
**Pull request by lmammino**
This change allows us to remap correctly the following types of links:

- `Plugins.md` -> `/docs/${version}/Plugins.md`
- `./Plugins.md` -> `/docs/${version}/Plugins.md`
 - `Plugins.md#some-thi_ng` -> `/docs/${version}/Plugins.md#some-thi_ng`
 - `./Plugins.md#some-thi_ng` -> `/docs/${version}/Plugins.md#some-thi_ng`

Checkout the regex on regex101:

[Regex validation](https://regex101.com/r/vJmF35/1)

https://regex101.com/r/vJmF35/1

Closes #182

### CODE REVIEW

**Comment by lmammino** [comment]
Preview for this build is here: https://observant-net.surge.sh/

Please help with validating that all the links are remapped as expected

**Review by ghost — APPROVED** [review]

**Comment by ghost** [comment]
@lmammino it seems to work correctly, at least I can access earlier broken links on the preview website.

Though if you'll try to open `Server` -> `addSchema` -> `To learn more, read the Validation and Serialization documentation.` (https://observant-net.surge.sh/docs/latest/Server/#addschema) it'll throw `404`. Maybe it would be nice to review the docs in the `fastify` repo. Because only the part of the errors was fixed.

**Review by mcollina — APPROVED** [review]
lgtm

File: src/scripts/processReleases.js [hunk 1]
```diff
@@ -155,13 +155,16 @@ function remapLinks (content, item) {
   const ecosystemLinkRx = /\(\/docs\/[\w\d.-]+\/Ecosystem\.md\)/gi
   const docInternalLinkRx = /\(\/docs\/[\w\d.-]+\/[\w\d-]+(.md)/gi
   const pluginsLink = /\(Plugins.md\)/gi
+  const relativeLinks = /\((.\/)?((\w+).md(#[a-z0-9\-_]+)?)\)/gi
```
Comment (Eomm) [inline src/scripts/processReleases.js]: This should fix the link spotted by @iandrc

```suggestion
  const relativeLinks = /\((.\/)?(([a-zA-Z0-9\-_]+).md(#[a-z0-9\-_]+)?)\)/gi
```

Comment (lmammino) [reply src/scripts/processReleases.js]: Awesome catch! Thanks @Eomm and thanks @iandrc for spotting this!

**Comment by lmammino** [comment]
@iandrc + @Eomm I updated the preview based on your feedback and that issue seems to be solved! 🥳

https://observant-net.surge.sh/docs/latest/Server/#addschema

**Comment by lmammino** [comment]
I will go ahead and get this merged for now, we can iterate over and fix more stuff later if needed!

Good news is that this should also fix the documentation for old versions.

In the future, I'll try to see if there's any utility that can help us to identify 404 pages so that we can fix all the issues in a comprehensive way. In this sense, any advice is most welcome :)

**Pull request merged**